### PR TITLE
Migrate to version `0.6.0` of the `fastly` crate

### PR DESCRIPTION
**Pull request by doramatadora**
Upgrades the default Compute@Edge starter kit for rust to version `0.6.0` of the `fastly` crate, along with:
- a more descriptive version of the rust-toolchain file
- minor copyedits for consistency with Developer Hub and other starter kits
- linting
- adding Rust SME team as codeowners

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -131,17 +120,15 @@ impl LogLine {
...
 /// non-cacheable via cache-control header directives and adds appropriate CORS
 /// headers required for the beacon preflight request.
-pub fn generate_no_content_response() -> Result<Response<Body>, Error> {
-    Ok(Response::builder()
-        .status(StatusCode::NO_CONTENT)
-        .header(header::CONTENT_TYPE, "application/json")
-        .header(
+pub fn generate_empty_204_response() -> Response {
+    Response::from_status(StatusCode::NO_CONTENT)
+        .with_header(header::CONTENT_TYPE, "application/json")
```
[the remark this answers is not in the document]
Comment (jeddenlea) [reply src/main.rs:125]: Nevermind, I see https://httpwg.org/specs/rfc7231.html#rfc.section.6.3.5 implies Content-Type may be appropriate to set.

File: src/main.rs [hunk 2]
```diff
@@ -29,21 +26,19 @@ use serde_json::value::Value as ReportBody;
...
+            let _ = handle_reports(req)?;
+            // Return an empty 204 response to the downstream client.
+            Ok(generate_empty_204_response())
         }
-        // For all other requests return a 404 not found.
-        _ => Ok(Response::builder()
-            .status(StatusCode::NOT_FOUND)
-            .body(Body::from("Not found"))?),
+        // For all other requests return a 404.
+        _ => Ok(Response::from_body("Not found").with_status(StatusCode::NOT_FOUND)),
```
Comment (jeddenlea) [inline src/main.rs:41]: Apologies, I thought I had written this into my last review.

Plain text responses should generally have trailing newlines.  And we also have a handy helper function for such cases which sets an appropriate `Content-Type: text/plain; charset=utf-8`. (this is contrary to the docs which incorrectly say it sets "text/html", and that is going to be fixed).

```suggestion
        _ => Ok(Response::from_status(StatusCode::NOT_FOUND).with_body_str("Not found\n")),
```

Comment (doramatadora) [reply src/main.rs:41]: The newline at the end seems counter-intuitive. I wonder if this should be left to the function implementation.

**Review by kailan — APPROVED** [review]
Looks good to me! Only one minor comment about versioning ✨

File: Cargo.toml [hunk 3]
```diff
@@ -1,15 +1,15 @@
 [package]
 name = "compute-starter-kit-rust-beacon-termination"
 version = "0.2.0"
-authors = ["Dora Militaru <dmilitaru@fastly.com>"]
+authors = []
 edition = "2018"
 
 [profile.release]
 debug = true
```
Comment (triblondon) [inline Cargo.toml:8]: Think we are typically removing these?

File: src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (triblondon) [reply src/main.rs:125]: I'd typically expect to not see a content-type here.  Also, we are using this function to generate responses to CORS preflight OPTIONS requests, for which an application/json type doesn't seem to make sense.

**Review by triblondon — APPROVED** [review]
My comments are non-blocking suggestions.  LGTM!

File: src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (doramatadora) [reply src/main.rs:125]: Agreed, @triblondon – I'll remove this.

File: Cargo.toml (hunk 3 above)
Comment (doramatadora) [reply Cargo.toml:8]: Really? The default starter kit still has it.

**Pull request merged**